### ISSUE TITLE
[Refactor] Improve SEO Optimization and accessibility

### ISSUE THREAD

**Issue by Anmol-Baranwal** [priyankarpal/ProjectsHut#1943]
### File Name

Multiple Components

### Reason for Refactoring the Code

- Convert `<a>` anchor tag to `<Link>` component.

Reference Docs: https://nextjs.org/docs/pages/api-reference/components/link

- Add `aria-label` attribute to every link and button in the codebase

The `aria-label` attribute is an essential accessibility feature in HTML. It is used to provide a descriptive label for elements that might not have visible text or when the existing text content is not sufficient to convey the element's purpose to users of assistive technologies (e.g., screen readers).

In short, adding `aria-label` helps provide context to users who rely on assistive technologies.

### Record

- [X] I have checked the existing [issues](https://github.com/priyankarpal/ProjectsHut/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/priyankarpal/ProjectsHut/blob/main/contributing.md)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/priyankarpal/ProjectsHut/blob/main/CODE_OF_CONDUCT.md)
- [X] I'm a GSSoC'23 contributor
- [X] I want to work on this issue

**Comment by priyankarpal** [comment]
can you send me the file? (for `a` tag)

**Comment by Anmol-Baranwal** [comment]
> can you send me the file? (for `<a>` tag)

![image](https://github.com/priyankarpal/ProjectsHut/assets/74038190/5b694ca5-6bce-4961-b159-8e267f2ead07)

**Comment by priyankarpal** [comment]
> > can you send me the file? (for `<a>` tag)
> 
> ![image](https://user-images.githubusercontent.com/74038190/258082800-5b694ca5-6bce-4961-b159-8e267f2ead07.png)

got it.

**Comment by Anmol-Baranwal** [comment]
@priyankarpal 
I want to clarify that I have no intention of arguing.

In the open-source community, I've witnessed instances where a single-word contribution to a repository fixed major issues for over 1 million users. Therefore, it's important not to judge the significance of contributions solely based on the amount of code, but rather the impact they create.

For example, the `<Link>` component might be unfamiliar to some developers unless they've thoroughly explored the documentation or worked with recent versions of the framework. Hence, even proposing a minor improvement can have a substantial impact.

Considering this, I would recommend adding level labels after reviewing the pull request.

> Thanks for the GPG key. I have successfully implemented it.

**Comment by priyankarpal** [comment]
@Anmol-Baranwal I completely agree with your perspective on the significance of contributions in the open-source community. 

But can you please clarify this? 
> Considering this, I would recommend adding level labels after reviewing the pull request.

**Comment by Anmol-Baranwal** [comment]
> @Anmol-Baranwal I completely agree with your perspective on the significance of contributions in the open-source community.
> 
> But can you please clarify this?
> 
> > Considering this, I would recommend adding level labels after reviewing the pull request.

I believe adding levels of PR (as labels) after the PR is more appropriate. That is just my opinion😅

**Comment by priyankarpal** [comment]
> > @Anmol-Baranwal I completely agree with your perspective on the significance of contributions in the open-source community.
> > But can you please clarify this?
> > > Considering this, I would recommend adding level labels after reviewing the pull request.
> 
> I believe adding levels of PR (as labels) after the PR is created is more appropriate. That is just my opinionsweat_smile
> 
> Anyway, let me know the changes in the PR.

oh! got it
